File: web/src/types/localKnowledge.ts
export interface LocalKnowledgeImportRow {
  title?: string
  text: string
  symbol?: string
  as_of_date?: string
  tags?: string[]
}

export interface LocalKnowledgeImportValidationRequest {
  source_label: string
  default_symbol?: string
  rows: LocalKnowledgeImportRow[]
}

export interface LocalKnowledgeImportConfirmRequest extends LocalKnowledgeImportValidationRequest {
  import_batch_id: string
  confirm_reason: string
}

export interface LocalKnowledgeImportRisk {
  code: string
  severity: 'warning' | 'blocking'
  message: string
}

export interface LocalKnowledgeImportRowResult {
  row_number: number
  status: 'valid' | 'warning' | 'blocking'
  symbol: string
  title_preview: string
  text_preview: string
  content_hash: string
  risks: LocalKnowledgeImportRisk[]
}

export interface LocalKnowledgeImportIndexPlan {
  rag_chunk_count: number
  index_status: string
}

export interface LocalKnowledgeImportValidationSummary {
  total_count: number
  valid_count: number
  warning_count: number
  blocking_count: number
}

export interface LocalKnowledgeImportValidationResponse {
  import_batch_id: string
  summary: LocalKnowledgeImportValidationSummary
  rows: LocalKnowledgeImportRowResult[]
  index_plan: LocalKnowledgeImportIndexPlan
  safety_note: string
}

export interface LocalKnowledgeImportConfirmResponse {
  import_batch_id: string
  intelligence_item_count: number
  summary_count: number
  rag_chunk_count: number
  verification_count: number
  audit_event_ids: string[]
  index_status: string
  safety_note: string
}

File: internal/application/dto/local_knowledge.go
package dto

type LocalKnowledgeImportRow struct {
	Title     string   `json:"title"`
	Text      string   `json:"text"`
	Symbol    string   `json:"symbol,omitempty"`
	SourceURL string   `json:"source_url,omitempty"`
	AsOfDate  string   `json:"as_of_date,omitempty"`
	Tags      []string `json:"tags,omitempty"`
}

type LocalKnowledgeImportValidationRequest struct {
	SourceLabel   string                    `json:"source_label"`
	DefaultSymbol string                    `json:"default_symbol,omitempty"`
	Rows          []LocalKnowledgeImportRow `json:"rows"`
}

type LocalKnowledgeImportConfirmRequest struct {
	ImportBatchID string                    `json:"import_batch_id"`
	ConfirmReason string                    `json:"confirm_reason"`
	SourceLabel   string                    `json:"source_label"`
	DefaultSymbol string                    `json:"default_symbol,omitempty"`
	Rows          []LocalKnowledgeImportRow `json:"rows"`
}

type LocalKnowledgeImportRisk struct {
	Code     string `json:"code"`
	Severity string `json:"severity"`
	Message  string `json:"message"`
}

type LocalKnowledgeImportRowResult struct {
	RowNumber      int                        `json:"row_number"`
	Status         string                     `json:"status"`
	Symbol         string                     `json:"symbol,omitempty"`
	TitlePreview   string                     `json:"title_preview"`
	TextPreview    string                     `json:"text_preview"`
	ContentHash    string                     `json:"content_hash"`
	EstimatedChunk int                        `json:"estimated_chunk_count"`
	Risks          []LocalKnowledgeImportRisk `json:"risks"`
}

type LocalKnowledgeImportIndexPlan struct {
	RAGChunkCount int    `json:"rag_chunk_count"`
	IndexStatus   string `json:"index_status"`
}

type LocalKnowledgeImportValidationSummary struct {
	TotalCount    int `json:"total_count"`
	ValidCount    int `json:"valid_count"`
	BlockingCount int `json:"blocking_count"`
	WarningCount  int `json:"warning_count"`
}

type LocalKnowledgeImportValidationResponse struct {
	ImportBatchID string                                `json:"import_batch_id"`
	Summary       LocalKnowledgeImportValidationSummary `json:"summary"`
	Rows          []LocalKnowledgeImportRowResult       `json:"rows"`
	IndexPlan     LocalKnowledgeImportIndexPlan         `json:"index_plan"`
	SafetyNote    string                                `json:"safety_note"`
}

type LocalKnowledgeImportConfirmResponse struct {
	ImportBatchID         string   `json:"import_batch_id"`
	IntelligenceItemCount int      `json:"intelligence_item_count"`
	SummaryCount          int      `json:"summary_count"`
	RAGChunkCount         int      `json:"rag_chunk_count"`
	VerificationCount     int      `json:"verification_count"`
	IndexStatus           string   `json:"index_status"`
	AuditEventIDs         []string `json:"audit_event_ids"`
	SafetyNote            string   `json:"safety_note"`
}
